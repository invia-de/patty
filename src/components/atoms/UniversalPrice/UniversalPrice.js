import React from 'react';
import PropTypes from 'prop-types';
import Price from '../../utilities/Price/Price';
import url from '../../../utils/url';
import './Universalprice.module.scss';

export default function UniversalPrice({
  priceInEuro,
  price,
  currency,
  priceTotalInEuro,
  priceTotal
}) {
  const usePriceTotal =
    url.get('crm') !== false ||
    (url.get('children') !== false && url.get('children') > 0);

  if ((usePriceTotal && !priceTotal) || (currency && !priceTotal)) {
    return null;
  }

  return (
    <div>
      {currency !== 'EUR' && (
        <div>
          <Price
            value={usePriceTotal ? priceTotal : price}
            currency={currency}
          />
        </div>
      )}
      <Price value={usePriceTotal ? priceTotalInEuro : priceInEuro} />
    </div>
  );
}
UniversalPrice.propTypes = {
  priceInEuro: PropTypes.number,
  price: PropTypes.number,
  currency: PropTypes.string,
  priceTotalInEuro: PropTypes.number,
  priceTotal: PropTypes.number
};
UniversalPrice.defaultProps = {
  priceInEuro: 0,
  price: 0,
  currency: 'EUR',
  priceTotalInEuro: 0,
  priceTotal: 0
};
