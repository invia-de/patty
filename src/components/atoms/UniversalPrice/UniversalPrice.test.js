import React from 'react';
import UniversalPrice from './UniversalPrice';
import { render } from 'react-testing-library';

test('UniversalPrice renders', () => {
  const { container } = render(<UniversalPrice />);
  expect(container.firstChild).toMatchSnapshot();
});
