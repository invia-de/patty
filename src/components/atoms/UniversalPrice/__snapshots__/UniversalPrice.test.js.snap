// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversalPrice renders 1`] = `null`;
